network/src/services/outbound_peer.rs

```diff
@@ -55,8 +55,23 @@ impl OutboundPeerService {
 
     fn dial_feeler(&mut self) {
         let now_ms = unix_time_as_millis();
+        let filter: Box<dyn Fn(&AddrInfo) -> bool> = match self.transport_type {
+            TransportType::Tcp => Box::new(|_| true),
+            TransportType::Ws => Box::new(|peer_addr: &AddrInfo| {
+                peer_addr
+                    .addr
+                    .iter()
+                    .any(|p| matches!(p, Protocol::Dns4(_) | Protocol::Dns6(_) | Protocol::Tcp(_)))
+            }),
+            TransportType::Wss => Box::new(|peer_addr: &AddrInfo| {
+                peer_addr
+                    .addr
+                    .iter()
+                    .any(|p| matches!(p, Protocol::Dns4(_) | Protocol::Dns6(_)))
+            }),
+        };
```

Comment:
```suggestion
        let filter = |peer_addr: &AddrInfo| {
            match self.transport_type {
                TransportType::Tcp => true,
                TransportType::Ws => peer_addr
                    .addr
                    .iter()
                    .any(|p| matches!(p, Protocol::Dns4(_) | Protocol::Dns6(_) | Protocol::Tcp(_))),
                TransportType::Wss => peer_addr
                    .addr
                    .iter()
                    .any(|p| matches!(p, Protocol::Dns4(_) | Protocol::Dns6(_))),
            }
        };
```

Reply:
This is not possible because the filter return value types of each branch are different, so the only way to unify them is to use the dyn trait.